#2425: A fix for ParsingException: Unexpected token EofSymbolId

Igor-Zlatomrezhev opened the pull request:
`ReplaceData` method used to raise following exception when the worksheet had defined names:
>ClosedXML.Parser.ParsingException: 'Error at char 0 of '': Unexpected token EofSymbolId.'

Igor-Zlatomrezhev committed:
#2425: `ReplaceData` method used to raise following exception when the worksheet had defined names:
> ClosedXML.Parser.ParsingException: 'Error at char 0 of '': Unexpected token EofSymbolId.'

Igor-Zlatomrezhev committed:
#2425: typo fixed

Igor-Zlatomrezhev committed:
#2425: one more unit test added

hevilhuy commented:
Someone please review this because it fixes an important problem with table of the latest version

heyixin79 commented:
@Igor-Zlatomrezhev @hevilhuy 
The same exception was raised when I use sheet.Row().InsertRowsBelow() and sheet.Column().InsertColumnsAfter().
When I tested f000ad4,
The ReplaceData() method and sheet.Row().InsertRowsBelow() method was OK, 
But sheet.Column().InsertColumnsAfter() still raised the same exception.
It seems that there is a same issue in MoveDefinedNamesColumns  method.

jahav committed:
Decuplicate tests

The two tests differ only by a formula of defined name. The test should
be more comprehensive (check defined names are shifted), but whole
formula shifting on structural changes is in the state of flux.

Remove AdjustToContents because it's unnecessary code for the test.
